### ISSUE TITLE
[explicit-function-return-type] Decorators being reported ESLint: Missing return type on function

### ISSUE THREAD

**Issue by abrarazeem** [typescript-eslint/typescript-eslint#2842]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
      "@typescript-eslint/explicit-function-return-type": ["error", {
      allowExpressions: true
    }]
  }
}
```

```TS
import { Controller, HttpCode, HttpStatus } from '@nestjs/common'
import { AppService } from './app.service'
// Actually it's imported from @nestjs/common package but for the sake of testing and quick glimpse adding here.
export declare const Get: (path?: string | string[]) => MethodDecorator

@Controller()
export class AppController {

  constructor(private readonly appService: AppService) {}

  @Get() // ESLint: Missing return type on function.(@typescript-eslint/explicit-function-return-type)
  @HttpCode(HttpStatus.OK)
  getHello() {

    return this.appService.getHello()
  }
}
```

**Expected Result**
Decorators shouldn't be reported as missing return type.

**Actual Result**

```
Missing return type on function.eslint@typescript-eslint/explicit-function-return-type
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.9.0` |
| `@typescript-eslint/parser`        | `4.9.0` |
| `TypeScript`                       | `4.1.2` |
| `ESLint`                           | `7.14.0` |
| `node`                             | `14.15.1` |

**Comment by bradzacher** [comment]
![image](https://user-images.githubusercontent.com/7462525/100963567-7926cc00-34db-11eb-92e1-0d72a82e27f2.png)

The error is higlighting the function, not the decorator.
`getHello` has no return type.
